pkg/asset/installconfig/azure/session.go

```diff
@@ -0,0 +1,155 @@
+package azure
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/Azure/go-autorest/autorest"
+	azureenv "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/AlecAivazis/survey.v1"
+)
+
+var authFileLocation = os.Getenv("HOME") + "/.azure/osServicePrincipal.json"
+
+//Session is an object representing session for subscription
+type Session struct {
+	SubscriptionID string
+	Authorizer     autorest.Authorizer
+}
+
+// GetSession returns an azure session by checking credentials
+// and, if no creds are found, asks for them and stores them on disk in a config file
+func GetSession() (*Session, error) {
+	err := getCredentials()
+	if err != nil {
+		return nil, err
+	}
+	return newSessionFromFile()
+}
+
+func newSessionFromFile() (*Session, error) {
+	os.Setenv("AZURE_AUTH_LOCATION", authFileLocation)
```

Comment:
we should allow people to provide their own `AZURE_AUTH_LOCATION` and only default to `authFileLocation` when empty.

Reply:
yes, but is that really important to reach an MVP asap?

Reply:
resolving as won't change in this iteration